src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+            {
+                entityRestPath = (string)entityRestSettings;
+                if (!string.IsNullOrEmpty(entityRestPath) && entityRestPath.StartsWith('/'))
+                {
+                    // Remove slash from start of rest path.
+                    entityRestPath = entityRestPath.Substring(1);
+                }
+            }
+
+            Assert.IsFalse(Equals('/', entityRestPath));
+            return entityRestPath;
+        }
+
+        /// <summary>
+        /// Creates the base OpenApiResponse object common to all requests where
+        /// responses are of type "application/json".
+        /// </summary>
+        /// <param name="description">HTTP Response Code Name: OK, Created, BadRequest, etc.</param>
+        /// <param name="responseObjectSchemaName">Schema used to represent response records.
+        /// Null when an example (such as error codes) adds redundant verbosity.</param>
+        /// <returns>Base OpenApiResponse object</returns>
+        private static OpenApiResponse CreateOpenApiResponse(string description, string? responseObjectSchemaName = null)
+        {
+            OpenApiResponse response = new()
+            {
+                Description = description
+            };
+
+            // No entityname means no response object schema should be included.
+            // the entityname references the schema of the response object.
+            if (!string.IsNullOrEmpty(responseObjectSchemaName))
+            {
+                Dictionary<string, OpenApiMediaType> contentDictionary = new()
+                {
+                    {
+                        MediaTypeNames.Application.Json,
+                        CreateResponseContainer(responseObjectSchemaName)
+                    }
+                };
+                response.Content = contentDictionary;
+            }
+
+            return response;
+        }
+
+        /// <summary>
+        /// Creates the OpenAPI description of the response payload, excluding the result records:
+        /// {
+        ///     "value": [
+        ///         {
+        ///             "resultProperty": resultPropertyValue
+        ///         }
+        ///     ]
+        /// }
+        /// </summary>
+        /// <param name="responseObjectSchemaName">Schema name of response payload.</param>
+        /// <returns>The base response object container.</returns>
+        private static OpenApiMediaType CreateResponseContainer(string responseObjectSchemaName)
+        {
+            // schema for the response's collection of result records
+            OpenApiSchema resultCollectionSchema = new()
+            {
+                Reference = new OpenApiReference()
+                {
+                    Type = ReferenceType.Schema,
+                    Id = $"{responseObjectSchemaName}"
+                }
+            };
+
+            // Schema for the response's root property "value"
```

Comment:
For pagination, should we add the property `nextLink` here?

Reply:
yes, good idea, i'll need to add that but have to make sure that it is only added for GET (all) requests which returns many items.

Reply:
added nextlink and response body example for getall looks like:
```json
{
  "value": [
    {
      "id": 0,
      "name": "string"
    }
  ],
  "nextLink": "string"
}
```